Lib/Src/CCmdBuffer.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * @file CFifoBuffer.cpp
+ */
+
+/*
+ *  Created on: Mar 28, 2022
+ *      Author: niuslar
+ */
+
+#include <CCmdBuffer.h>
+
+CCmdBuffer::CCmdBuffer() : m_head(0), m_tail(BUF_SIZE)
+{
+    // TODO Auto-generated constructor stub
+}
+
+CCmdBuffer::~CCmdBuffer()
+{
+    // TODO Auto-generated destructor stub
+}
+
+/**
+ * @brief Adds character to buffer
+ * @return True if the character is \n or \r to mark end of string
+ */
+bool CCmdBuffer::put(const char data)
```

Comment:
This is susceptible to buffer overwrite. By the time head resets the data at that position must already be read by the software. This may not always be the case. Have a look at how circular buffers / FIFO  works. They are everywhere in the real-world packaged under different names. So worth learning about it.